pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -129,6 +129,12 @@ func (i *KumaInjector) needInject(pod *kube_core.Pod, ns *kube_core.Namespace) (
 		log.V(1).Info("pod fulfills exception requirements")
 		return false, nil
 	}
+
+	// A pod with this annotation provides a built-in gateway
+	if builtin, builtinExists := metadata.Annotations(pod.Annotations).GetString(metadata.KumaBuiltinGatewayAnnotation); builtinExists && builtin != "" {
+		return false, nil
+	}
+
```

Comment:
This annotation also holds the name of the `Gateway` unfortunately.

Reply:
Right, but we don't need to overload that to also deter the injector, since we have a dedicated way  to do that.

Reply:
Ah ok, you mean just add the `sidecar-injection` annotation to the gateway pod instead of checking for `gateway` here. Makes sense, I thought you wanted to say we don't need the annotation itself.